# 00 -- Ambiente de trabalho e primeiro notebook

**GA033 · Método de Elementos Finitos: Implementação Computacional · LNCC**  
**Diego Tavares Volpatto · CC BY 4.0**

Este caderno é o primeiro contato com o ambiente. Não é necessário conhecer Python. Vamos executar algumas células, modificar um cálculo e entender onde os resultados ficam armazenados.

Ao terminar, você deverá conseguir abrir um notebook, executar e editar uma célula, reiniciar o kernel e reconhecer a função do Pixi.

## 1. Abrir o ambiente

Na pasta que contém `pixi.toml`, execute os comandos abaixo **no terminal**, uma linha por vez:

```sh
pixi install --locked
pixi run lab
```

Abra este arquivo na pasta `notebooks`. O Pixi prepara Python e bibliotecas; o JupyterLab apresenta o caderno no navegador. O código é executado por um processo chamado **kernel**. Veja o [guia do ambiente](../docs/ambiente-pixi.md) se precisar de ajuda com a instalação.

Para acompanhar as primeiras aulas, basta usar o ambiente fornecido. Voltaremos aos arquivos do Pixi depois de usar algumas bibliotecas.

## 2. Texto e código

Esta célula contém texto em Markdown. A próxima contém código Python. Clique nela e pressione **Shift+Enter**. O resultado aparece logo abaixo.

O que você espera que seja mostrado?

In [1]:
print("Olá! Vamos começar a programar em Python.")

Olá! Vamos começar a programar em Python.


`print` mostra uma informação. O texto entre aspas é uma string, assunto do próximo caderno. A combinação **Shift+Enter** executa a célula e passa para a próxima; **Ctrl+Enter** executa mantendo a seleção.

**Sua vez:** altere a mensagem acima, execute novamente e confira a saída.

## 3. Um primeiro cálculo

Vamos dividir um intervalo de comprimento 2 em quatro partes iguais. Antes de executar, calcule mentalmente o comprimento de cada parte.

In [2]:
comprimento = 2.0
numero_partes = 4
tamanho_parte = comprimento / numero_partes
print(tamanho_parte)

0.5


O resultado é `0.5`. Os nomes à esquerda de `=` guardam referências aos valores usados no cálculo. Os detalhes da atribuição serão estudados no caderno 01.

**Experimente:** mantenha o comprimento e use oito partes. Qual resultado você prevê? Modifique, execute e compare.

In [3]:
comprimento_exercicio = 2.0
numero_partes_exercicio = 4  # Altere este valor para experimentar.
print(comprimento_exercicio / numero_partes_exercicio)

0.5


<details>
<summary>Conferência depois de tentar</summary>

Com oito partes, `2.0 / 8` resulta em `0.25`. Ao dobrar o número de partes, o tamanho de cada parte cai pela metade, mantendo o comprimento total.

</details>

## 4. O kernel lembra o que foi executado

As variáveis continuam disponíveis entre células **do mesmo kernel**. A próxima célula usa um valor definido antes.

In [4]:
print("Comprimento informado:", comprimento)

Comprimento informado: 2.0


A ordem visual das células e a ordem de execução podem ser diferentes. Se você mudar um valor em uma célula e não executar novamente o cálculo que depende dele, a saída antiga continuará aparecendo.

Para conferir o caderno como um programa completo, use **Kernel → Restart Kernel and Run All Cells** (reiniciar o kernel e executar todas as células). O nome do menu pode variar com a interface e o idioma. Isso apaga as variáveis em memória e executa o código de cima para baixo.

**Sua vez:** reinicie e execute tudo. O caderno deve funcionar sem depender de algo digitado em outra sessão.

## 5. Salvar não é executar

Salvar o arquivo preserva o texto, o código e as saídas registradas no notebook. Não salva todas as variáveis que existem na memória do kernel. Ao reabrir o material, execute as células para reconstruir o cálculo.

Use **Ctrl+S** (ou **Cmd+S** no macOS) para salvar. Na aula de Git veremos que salvar um arquivo também é diferente de registrar uma versão dele no histórico.

## 6. Conferir o Python usado

O comando abaixo serve como diagnóstico do ambiente. `import` permite usar funcionalidades de um módulo; vamos estudar essa ideia aos poucos.

In [5]:
import sys

print("Versão do Python:", sys.version.split()[0])

Versão do Python: 3.13.16


O ambiente do curso especifica Python 3.13. A versão exata instalada é registrada em `pixi.lock`. Se aparecer outra versão, confira o kernel selecionado e retome as instruções do ambiente.

Não é preciso instalar pacotes dentro das células. A instalação fica descrita no projeto e é realizada pelo Pixi.

## 7. Terminal ou célula?

| O que você quer fazer | Onde escrever |
| --- | --- |
| Iniciar o Jupyter com `pixi run lab` | Terminal, na pasta do projeto |
| Calcular `2.0 / 4` | Célula de código Python |
| Explicar o objetivo de um cálculo | Célula Markdown |
| Adicionar uma biblioteca com `pixi add` | Terminal; faremos isso em um projeto de prática |

**Atividade final:** escreva uma frase explicando a diferença entre Pixi, Jupyter e Python. Depois, crie uma célula de código que calcule a área de um retângulo de lados 3 e 2.

In [6]:
base = 3.0
altura = 2.0
# Escreva abaixo a expressão que calcula e mostra a área.

<details>
<summary>Uma solução possível</summary>

```python
area = base * altura
print(area)
```

O resultado é `6.0`. Pixi gerencia o ambiente, Jupyter apresenta e organiza o caderno, e o interpretador Python executa o código.

</details>

## Próximo passo

Continue em [01 -- Primeiros passos](01-primeiros-passos.ipynb). Para encerrar o Jupyter, volte ao terminal que o iniciou, pressione **Ctrl+C** e confirme se solicitado.

Referências: [JupyterLab](https://jupyterlab.readthedocs.io/en/stable/user/notebook.html), [Pixi](https://pixi.prefix.dev/latest/first_workspace/).

[Índice do curso](../README.md) · [Próximo caderno →](01-primeiros-passos.ipynb)

Autor: Diego Tavares Volpatto · [CC BY 4.0](https://creativecommons.org/licenses/by/4.0/deed.pt-br)